Microsoft.Toolkit/Diagnostics/Generated/ThrowHelper.Collection.tt

```diff
@@ -13,9 +13,9 @@ using Microsoft.Toolkit.Extensions;
 namespace Microsoft.Toolkit.Diagnostics
 {
     /// <summary>
-    /// Helper methods to throw exceptions
+    /// Helper methods to efficiently throw exceptions.
     /// </summary>
-    internal static partial class ThrowHelper
+    public static partial class ThrowHelper
```

Comment:
I'm actually wondering do we just keep these all internal and call this `GuardThrowHelper` and then have the new generalized `ThrowHelper` be the sole public one, would make it easier to understand the connections?

Reply:
Ah, I see now, it still uses the internal structure of the ThrowHelper helpers...

Maybe instead we put all the partials that are `internal` only methods in an `internal` sub-folder? That way it's clear which files contribute to the public API?

@azchohfi what are your thoughts here on this?

Reply:
I like the idea of moving all the APIs that are internal into an "Internal" folder, that'd make all the APIs in the main folder public, and all the others (or the generated ones) would be either in the "Generated" folder or in this new one 😄

Reply:
Made a refactoring in https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3346/commits/7c7098ada9d7e3f06c59f3afd2b722311556e236, I used the same `"Internals"` folder name we're also using in the `HighPerformance` package, and renamed the different files to make it clearer which `Guard` file they contain internal APIs for. I think the structure of the `Diagnostics` namespace is much easier to follow now, let me know if that works for you! 😊

Reply:
Much better!